### ISSUE TITLE
how to separate training environments and evaluation environments

### ISSUE THREAD

**Issue by amineoui** [opendilab/LightZero#112]
how we can separate training environment and evaluation environment , the goal of doing that is separate the training data and evaluation data during the training.
there is a way to give 2 env on load as train_env and eval_env ?
or something can help me like a variable or method that check if the loaded environment is for eval or train as custom gym env

thank you for helping

**Comment by puyuan1996** [comment]
- Hello, our current framework employs essentially the same environment for data collection and evaluation stages, such as `Pong`, but also allows for fine-tuning the environment in these two stages. 

- Specifically, as you can see in [here](https://github.com/opendilab/LightZero/blob/main/zoo/atari/envs/atari_lightzero_env.py#L168), we can set different parameters for each stage. For example, in the data collection stage, we might use clipped rewards, while in the evaluation stage, we display the original rewards. This can be achieved by setting the `clip_rewards` parameter differently. The purpose of this design is to enhance the flexibility and adaptability of our model to meet a variety of experimental needs. 

- However, we currently do not support using completely different environments for the collect and eval stages, which so far is not necessary in practice. 

- To determine whether the current environment is for collect or eval, you can simply set a judgment flag [here](https://github.com/opendilab/LightZero/blob/main/zoo/atari/envs/atari_lightzero_env.py#L168), such as `is_collect`.

**Comment by amineoui** [comment]
thank you , is clear about is_collect=True under create_evaluator_env_cfg , but i dont get where it called the create_evaluator_env_cfg function on the framework. 

also please tell me more about clip_rewards how it work and there effect , and i should change on my reward function ?

im gotting mistiruse problem train is going well but evaluation with new data is so so so bad .
like is learn only train data.

thank you so much

**Comment by amineoui** [comment]
sorry i just update to last version i was in old version of the framework , thank you for your hard working on this art of RL really a great job. i appreciate that.

please tell me more about clip_rewards and there effect , and how to make the agent flexible and adaptable with new data
is i should just do the function like this on my custom env , or i should activate some hyperparams on the config file ?
```
    @staticmethod
    def create_evaluator_env_cfg(cfg: dict) -> List[dict]:
        evaluator_env_num = cfg.pop('evaluator_env_num')
        cfg = copy.deepcopy(cfg)
        cfg.max_episode_steps = cfg.eval_max_episode_steps
        cfg.episode_life = False
        cfg.clip_rewards = False
        cfg.is_collect = False
        return [cfg for _ in range(evaluator_env_num)]

```

**Comment by puyuan1996** [comment]
Thank you for your support and encouragement!

Regarding ``clip_rewards``, it is a commonly used technique to limit the range of reward values in reinforcement learning. In many cases, reward values can have a wide variance, which can lead to instability in training or the problem of gradient explosion. The role of clip_rewards is to constrain reward values within a smaller range, usually [-1, 1]. In the Atari environment, the original reward is converted into one of three values, `{-1,0,1}`, based on its sign. You can refer to this [implementation](https://github.com/opendilab/DI-engine/blob/main/ding/envs/env_wrappers/env_wrappers.py#L238) for more details. This helps control the size of the gradient, making the training process more stable. To gain more insights into the robustness of reinforcement learning to reward scale, you can refer to the following papers:  [paper1](https://arxiv.org/abs/1602.07714), [paper2](https://arxiv.org/abs/1805.11593).

Regarding the issue of ``enhancing the flexibility and stability of agents``, this is a broad question. You can refer to papers in related fields such as **Exploration-Exploitation Balance**, **Transfer Learning**, **Curriculum Learning**, and **Domain Randomization**.

If you have any questions, feel free to raise them for discussion. Best wishes.
